server/covmanager/tasks.py

```diff
@@ -22,3 +26,39 @@ def check_revision_update(pk):
     # reject the client anymore.
 
     return
+
+
+@app.task
+def aggregate_coverage_data(pk, pks):
+    from covmanager.models import Collection, CollectionFile  # noqa
+    from FTB import CoverageHelper # noqa
+
+    # Fetch our existing, but incomplete destination collection
+    mergedCollection = Collection.objects.get(pk=pk)
+
+    # Fetch all source collections
+    collections = Collection.objects.filter(pk__in=pks)
+
+    # Merge the coverage of all other collections into the first one
+    first_collection = collections[0]
+    first_collection.loadCoverage()
+    newCoverage = first_collection.content
+    for collection in collections[1:]:
+        # Load coverage, perform the merge, then release reference to the JSON blob again
+        collection.loadCoverage()
+        CoverageHelper.merge_coverage_data(newCoverage, collection.content)
+        collection.content = None
+
+    # Save the new coverage blob to disk and database
+    newCoverage = json.dumps(newCoverage, separators=(',', ':'))
+    h = hashlib.new('sha1')
+    h.update(repr(newCoverage).encode('utf-8'))
```

Comment:
Using `repr()` on this seems unnecessary. We just got it from `json.dumps()`, so we know it's a string.